core/src/main/java/org/apache/iceberg/BaseSnapshot.java

```diff
@@ -50,38 +52,42 @@
   /**
    * For testing only.
    */
-  BaseSnapshot(TableOperations ops,
-               long snapshotId,
-               String... manifestFiles) {
-    this(ops, snapshotId, null, System.currentTimeMillis(), null, null,
+  BaseSnapshot(FileIO io,
+      Map<Integer, PartitionSpec> specsById,
+      long snapshotId,
+      String... manifestFiles) {
+    this(io, snapshotId, null, System.currentTimeMillis(), null, null,
         Lists.transform(Arrays.asList(manifestFiles),
-            path -> new GenericManifestFile(ops.io().newInputFile(path), 0)));
+            path -> new GenericManifestFile(io.newInputFile(path), 0)), specsById);
   }
 
-  BaseSnapshot(TableOperations ops,
-               long snapshotId,
-               Long parentId,
-               long timestampMillis,
-               String operation,
-               Map<String, String> summary,
-               InputFile manifestList) {
-    this.ops = ops;
+  BaseSnapshot(FileIO io,
+      long snapshotId,
```

Comment:
Whitespace changes are making this touch more lines than necessary. Can you please revert them?